### ISSUE TITLE
_unsortedTagList - performance issue

### ISSUE THREAD

**Issue by xposure** [prime31/Nez#355]
Taking an extreme case here to amplify the problem. If you add 40k sprites on scene init, it causes 40k entries in to `_unsortedTagList` all pointing to tag 0.

This will later make `updateLists()` attempt to sort the same `tag[0]` 40k times over 40k sprites. so a loop of 40000x40000 which is sorting the same data each iteration.

```
			// sort our tagList if needed
			if( _unsortedTags.Count > 0 )
			{
				for( int i = 0, count = _unsortedTags.Count; i < count; i++ )
				{
					var tag = _unsortedTags[i];
                                        //tag is always 0 so its always sorting the same list 40k times
					_entityDict[tag].sort(); 
				}
				_unsortedTags.Clear();
			}
```

Shouldn't _unsortedTags be a hashset as you only want to record which tags need updated uniquely?

EDIT: I understand this is an extreme case and wouldn't do this in production but even simply adding 100 sprites would cause 99 redundant sorts, correct me if I'm wrong.

**Comment by xposure** [comment]
https://github.com/xposure/FFNez/commit/0e700011a4bca5a801841e42151beb98514c95e4

**Comment by prime31** [comment]
Closed by commit b99df26fb234b932c4fa60c1357b8c58f2d09b2b
